src/common/src/config.rs

```diff
@@ -790,22 +787,27 @@ pub struct SystemConfig {
     pub pause_on_next_bootstrap: Option<bool>,
 }
 
-impl SystemConfig {
-    pub fn into_init_system_params(self) -> SystemParams {
+impl RwConfig {
+    // NOTE(eric): The naming is confusing here. `SystemParams` means the paratmeters that are
+    // persisted in Meta store, while the `SystemConfig` in this file corresponds to the `[system]`
+    // section in `risingwave.toml`
+    pub fn as_initial_system_params(&self) -> SystemParams {
         SystemParams {
-            barrier_interval_ms: self.barrier_interval_ms,
-            checkpoint_frequency: self.checkpoint_frequency,
-            sstable_size_mb: self.sstable_size_mb,
-            parallel_compact_size_mb: self.parallel_compact_size_mb,
-            block_size_kb: self.block_size_kb,
-            bloom_false_positive: self.bloom_false_positive,
-            state_store: self.state_store,
-            data_directory: self.data_directory,
-            backup_storage_url: self.backup_storage_url,
-            backup_storage_directory: self.backup_storage_directory,
-            telemetry_enabled: self.telemetry_enabled,
-            max_concurrent_creating_streaming_jobs: self.max_concurrent_creating_streaming_jobs,
-            pause_on_next_bootstrap: self.pause_on_next_bootstrap,
+            barrier_interval_ms: self.system.barrier_interval_ms,
+            checkpoint_frequency: self.system.checkpoint_frequency,
+            sstable_size_mb: self.system.sstable_size_mb,
+            parallel_compact_size_mb: self.system.parallel_compact_size_mb,
+            block_size_kb: self.system.block_size_kb,
+            bloom_false_positive: self.system.bloom_false_positive,
+            state_store: self.system.state_store.clone(),
+            data_directory: self.system.data_directory.clone(),
+            backup_storage_url: self.system.backup_storage_url.clone(),
+            backup_storage_directory: self.system.backup_storage_directory.clone(),
+            max_concurrent_creating_streaming_jobs: self
+                .system
+                .max_concurrent_creating_streaming_jobs,
+            pause_on_next_bootstrap: self.system.pause_on_next_bootstrap,
+            telemetry_enabled: Some(self.server.telemetry_enabled),
```

Comment:
So we still have it persisted in the meta store? Is this intended? If so, should we instead remove the entry under `[server]`?

Reply:
> So we still have it persisted in the meta store? Is this intended?

I guess it's because #7384 wants to allow setting from SQL interface:

```
ALTER SYSTEM SET telemetry_enabled = false
```

This requires the parameter to be a system parameter instead of `RwConfig`.

> If so, should we instead remove the entry under `[server]`?

The value in `RwConfig` is used to initialize the settings when Meta node starts up for the first time, after it, the persisted value will be used. This pattern was also adopted by other parameters listed here, and it's very useful to set initial parameters for testing purpose.

Reply:
After some consideration, I feel that this is unfriendly to users. They don't know which parameters are only used to initialize meta parameters and which are not. Assume he disabled telemetry by set

```
[server]
telemetry_enabled = false
```

and then he changed the YAML to `telemetry_enabled = true`, this won't work because Meta will use the persisted value.

From this perspective, I'd like to keep `telemetry_enabled` under `[system]` section, because the implies `telemetry_enabled` is used for initialization. However, this is conflicted with the description:

```
/// The section `[system]` in `risingwave.toml`. This section is only for testing purpose and should
/// not be documented.
```

How about moving `telemetry_enabled` to `[system]` and declare `[system]` as "section for initializing system parameters"? cc. @hzxa21 @BugenZhao

Reply:
> How about moving telemetry_enabled to [system] and declare [system] as "section for initializing system parameters"? cc. @hzxa21 @BugenZhao

+1. I believe this is what it's originally for. 🥵

Reply:
Why does telemetry need to be `alter`ed by SQL 😇

Reply:
> After some consideration, I feel that this is unfriendly to users. They don't know which parameters are only used to initialize meta parameters and which are not. Assume he disabled telemetry by set
> 
> ```
> [server]
> telemetry_enabled = false
> ```
> 
> and then he changed the YAML to `telemetry_enabled = true`, this won't work because Meta will use the persisted value.
> 
> From this perspective, I'd like to keep `telemetry_enabled` under `[system]` section, because this implies `telemetry_enabled` is used for initialization. However, this is conflicted with the description:
> 
> ```
> /// The section `[system]` in `risingwave.toml`. This section is only for testing purpose and should
> /// not be documented.
> ```
> 
> How about moving `telemetry_enabled` to `[system]` and declare `[system]` as "section for initializing system parameters"? cc. @hzxa21 @BugenZhao 
> 

+1

Reply:
> Why does telemetry need to be `alter`ed by SQL 😇

I guess it just makes user easier to turn off telemetry by treating it as a mutable SQL system variable. I vaguely remember that we investigate how other systems do that and we have followed the TiDB way.